src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -347,6 +350,34 @@ public void DeleteDatabase(string dbName, DeletionInProgressStatus deletionInPro
                     {
                         configuration = CreateDatabaseConfiguration(dbName, ignoreDisabledDatabase: true, ignoreBeenDeleted: true, ignoreNotRelevant: true,
                             databaseRecord: record);
+
+                        using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                        using (context.OpenReadTransaction())
+                        {
+                            var allDatabasesRecords = _serverStore.Cluster.GetAllDatabases(context);
+                            var parentDirectory = configuration.Core.DataDirectory.FullPath;
+
+                            foreach (var currRecord in allDatabasesRecords)
+                            {
+                                if (currRecord.DatabaseName.Equals(dbName))
+                                    continue;
+
+                                var currConfiguration = CreateDatabaseConfiguration(currRecord.DatabaseName, ignoreDisabledDatabase: true, 
+                                    ignoreBeenDeleted: true, ignoreNotRelevant: true, databaseRecord: currRecord);
+
+                                var currDirectory = currConfiguration.Core.DataDirectory.FullPath;
+
+                                if (PathUtil.IsSubDirectory(currDirectory, parentDirectory))
+                                        throw new IntersectionInPaths($"Cannot delete database {dbName} from {parentDirectory}. " +
+                                                                      $"There is an intersection with database {currRecord.DatabaseName} located in {currDirectory}.");
+                            }
+                        }
+                    }
+                    catch (IntersectionInPaths e)
+                    {
+                        if (_logger.IsInfoEnabled)
+                            _logger.Info(e.Message);
+                        throw;
```

Comment:
Consider the ergonomics of this.  We already have error handling, why are you making it so?

Reply:
If I understand you correctly, the reason for adding the new catch is that I don't want the קxception thrown to reach:        
![image](https://user-images.githubusercontent.com/46426884/115678903-affe6100-a35a-11eb-9853-ba8e66e0655b.png)
(I want to reach directly the "finally" section). 
I removed the exception I created and change the exception to "InvalidOperationException", hope it's fine.

Reply:
Why don't you want it to reach this exception? What is the new error handling that you are implementing here?